api/clients/v2/verification/blob_verifier.go

```diff
@@ -0,0 +1,74 @@
+package verification
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/common"
+
+	disperser "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	verifierBindings "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDABlobVerifier"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// BlobVerifier is responsible for making eth calls against the BlobVerifier contract to ensure cryptographic and
+// structural integrity of V2 certificates
+type BlobVerifier struct {
+	// go binding around the EigenDABlobVerifier ethereum contract
+	blobVerifierCaller *verifierBindings.ContractEigenDABlobVerifierCaller
+}
+
+// NewBlobVerifier constructs a BlobVerifier
+func NewBlobVerifier(
+	ethClient *common.EthClient, // the eth client, which should already be set up
+	blobVerifierAddress string, // the hex address of the EigenDABlobVerifier contract
+) (*BlobVerifier, error) {
+
+	verifierCaller, err := verifierBindings.NewContractEigenDABlobVerifierCaller(
+		gethcommon.HexToAddress(blobVerifierAddress),
+		*ethClient)
+
+	if err != nil {
+		return nil, fmt.Errorf("bind to verifier contract at %s: %s", blobVerifierAddress, err)
+	}
+
+	return &BlobVerifier{
+		blobVerifierCaller: verifierCaller,
+	}, nil
+}
+
+// VerifyBlobV2FromSignedBatch calls the verifyBlobV2FromSignedBatch view function on the EigenDABlobVerifier contract
+//
+// This method returns nil if the blob is successfully verified. Otherwise, it returns an error.
```

Comment:
I prefer when a method returns (bool, error) for these kinds of cases, because there are many error cases that are unrelated to the cert being verified or not (networking error being the most prominent one).

Reply:
@samlaf The problem is that the underlying function call, `VerifyBlobV2FromSignedBatch`, communicates a failed verification via an error. So, there is never a case where this method would return `false, nil`, nor `true, notNil`

A returned bool _could_ be used to convey useful information, if we defined it to mean "returns true if there are no errors preceding verification". But a bool being returned from a "verify" function already has strong baked in expectations of meaning "verified/not verified"

Or, we could make the return signature `(bool, error)`, even though it's a bit redundant. LMK what you guys think

Reply:
Gotcha, that make sense. Was thinking the validation onchain was trivial, but I see it can already return many different types of errors, and it wouldn't make sense to compress those to a boolean result.